dht/addr.go

```diff
@@ -7,7 +7,7 @@ import (
 )
 
 // Used internally to refer to node network addresses.
-type dHTAddr interface {
+type DHTAddr interface {
```

Comment:
Rather than expose this performance hack type, when you pass addrs out of dht, just pass the *udp.Addr as a net.Addr.